### PR TITLE
add JoinBuilder to SelectBuilder

### PR DESCRIPTION
**Pull request by rnantes**
adds JoinBuilder to SelectBuilder enabling:
``` Swift
try self.db.select()
            .column("*")
            .from("planets")
            .join(method: .inner,
                  table: "galaxies",
                  from: SQLColumn("galaxyID", table: "planets"),
                  to: SQLColumn("id", table: "galaxies"))
            .all().wait()
```

### CODE REVIEW

File: Sources/SQLKit/Builders/SQLJoinBuilder.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+    /// Joins a table
+    ///
+    ///     builder.join(method: .inner
+    ///                  table: "galaxies",
+    ///                  from: SQLColumn("galaxyID", table: "planets"),
+    ///                  to: SQLColumn("id", table: "galaxys"))
+    ///
+    public func join(method: SQLJoinMethod, table: String, from: SQLColumn, to: SQLColumn) -> Self {
+        self.joins.append(SQLJoin.init(method: method,
```
Comment (tanner0101) [inline Sources/SQLKit/Builders/SQLJoinBuilder.swift]: No Xcode formatting please. You can indent like this:
```swift
foo(
    bar: "baz"
)
```

File: Sources/SQLKit/Builders/SQLJoinBuilder.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+                                       table: SQLTableIdentifier(table),
+                                       expression: SQLJoinBinaryExpression.init(from: from, to: to)))
```
Comment (tanner0101) [inline Sources/SQLKit/Builders/SQLJoinBuilder.swift]: Can we use just `SQLBinaryExpression` directly here?

File: Sources/SQLKit/Query/SQLTableIdentifier.swift [hunk 3]
```diff
@@ -0,0 +1,15 @@
+public struct SQLTableIdentifier: SQLExpression {
```
Comment (tanner0101) [inline Sources/SQLKit/Query/SQLTableIdentifier.swift]: This doesn't add anything beyond `SQLIdentifier`, so we should just use that directly.

File: Sources/SQLKit/Query/SQLJoinBinaryExpression.swift [hunk 4]
```diff
@@ -0,0 +1,19 @@
+public struct SQLJoinBinaryExpression: SQLExpression {
```
Comment (tanner0101) [inline Sources/SQLKit/Query/SQLJoinBinaryExpression.swift]: Same here, this doesn't add anything beyond what `SQLBinaryExpression` does.

File: Sources/SQLKit/Builders/SQLJoinBuilder.swift [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+extension SQLJoinBuilder {
+
+    /// Joins a table
+    ///
+    ///     builder.join(method: .inner
+    ///                  table: "galaxies",
+    ///                  from: SQLColumn("galaxyID", table: "planets"),
+    ///                  to: SQLColumn("id", table: "galaxys"))
+    ///
+    public func join(method: SQLJoinMethod, table: String, from: SQLColumn, to: SQLColumn) -> Self {
```
Comment (tanner0101) [inline Sources/SQLKit/Builders/SQLJoinBuilder.swift]: Maybe the `method` can default to `.inner`?

**Comment by tanner0101** [comment]
Merging latest master into your fork should fix the CI failures :)

**Comment by gwynne** [comment]
My utility methods in #84 added join support directly on `SQLSelectBuilder`. @tanner0101 Is this separate protocol a preferred implementation?

**Comment by tanner0101** [comment]
I believe some dialects allow joining on `DELETE`, right? In that case this protocol would be very helpful. Either way, separating into a protocol is nice for organization and readability.

**Comment by tanner0101** [comment]
@gwynne would you be interested in combining these changes with what #84 did?

**Comment by tanner0101** [comment]
I've implemented this against latest master here: https://github.com/vapor/sql-kit/pull/103

**Pull request closed without merging**